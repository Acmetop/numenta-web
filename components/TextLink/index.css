/**
 * Numenta shared web components
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

._visited:visited {
  color: #0277bd;
}

.textlink {
  composes: dim from 'tachyons';
  composes: link from 'tachyons';
  composes: underline-hover from 'tachyons';
  composes: _visited;
  color: #0277bd;
  word-wrap: normal;
}
