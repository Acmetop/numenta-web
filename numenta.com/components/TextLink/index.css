/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

._visited:visited {
  color: #0277bd;  /* color-blue-dark from '...theme' */
}

.textlink {
  composes: dim from 'tachyons';
  composes: link from 'tachyons';
  composes: underline-hover from 'tachyons';
  composes: color-blue-dark from '../../static/assets/css/theme.css';
  composes: _visited;
  word-wrap: normal;
}
