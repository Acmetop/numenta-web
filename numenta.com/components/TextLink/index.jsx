import {IndexLink, Link} from 'react-router'
import {prefixLink} from 'gatsby-helpers'  // eslint-disable-line import/no-unresolved, max-len
import React from 'react'

import styles from './index.css'


const TextLink = ({children, to}) => {
  const attrs = {
    // default internal non-index link
    className: styles.textlink,
    to: prefixLink(to),
  }
  let Node = Link

  if (to && to.match(/^.+:/)) {
    // external link (browser location)
    Node = 'a'
    attrs.href = to
    delete attrs.to
  }
  else if (to === '/') {
    // internal index main/home link
    Node = IndexLink
  }

  return (
    <Node {...attrs}>
      {children}
    </Node>
  )
}

TextLink.propTypes = {
  children: React.PropTypes.any.isRequired,
  to: React.PropTypes.string.isRequired,
}

export default TextLink
