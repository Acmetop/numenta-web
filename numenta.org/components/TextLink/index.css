/**
 * Numenta.org HTM Community website source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

._visited:visited {
  color: #0277bd;  /* color-blue-dark from '...theme' */
}

.textlink {
  composes: dim from 'tachyons';
  composes: link from 'tachyons';
  composes: underline-hover from 'tachyons';
  composes: color-blue-dark from '../../static/assets/css/theme.css';
  composes: _visited;
  word-wrap: normal;
}
