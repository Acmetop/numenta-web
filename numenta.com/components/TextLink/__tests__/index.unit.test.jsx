import React from 'react'
import renderer from 'react-test-renderer'

import TextLink from '../../TextLink'


describe('<TextLink /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <TextLink href="http://numenta.com">
        Numenta Homepage
      </TextLink>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
