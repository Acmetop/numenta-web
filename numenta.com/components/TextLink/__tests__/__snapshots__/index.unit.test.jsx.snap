exports[`TextLink React component Renders Augmented correctly 1`] = `
<a
  className="textlink"
  href="http://numenta.com"
  onClick={[Function instrumentOnClick]}
  target="_blank">
  Numenta Homepage
</a>
`;

exports[`TextLink React component Renders Default correctly 1`] = `
<a
  className="textlink"
  href="http://numenta.com"
  onClick={[Function instrumentOnClick]}
  target={undefined}>
  Numenta Homepage
</a>
`;
